src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Input/Commands/OrderingHelper.cs

```diff
@@ -0,0 +1,345 @@
+        }
+
+        /// <summary>
+        /// Move the project tree up over one of its siblings.
+        /// </summary>
+        public static Task<bool> TryMoveUpAsync(ConfiguredProject configuredProject, IProjectTree projectTree)
+        {
+            Requires.NotNull(configuredProject, nameof(configuredProject));
+            Requires.NotNull(projectTree, nameof(projectTree));
+
+            return TryMoveAsync(configuredProject, projectTree, MoveAction.Up);
+        }
+
+        /// <summary>
+        /// Checks if the given project tree can move down over one of its siblings.
+        /// </summary>
+        public static bool CanMoveDown(IProjectTree projectTree)
+        {
+            Requires.NotNull(projectTree, nameof(projectTree));
+
+            return GetSiblingByMoveAction(projectTree, MoveAction.Down) != null;
+        }
+
+        /// <summary>
+        /// Move the project tree down over one of its siblings.
+        /// </summary>
+        public static bool TryMoveDown(Project project, IProjectTree projectTree)
+        {
+            Requires.NotNull(project, nameof(project));
+            Requires.NotNull(projectTree, nameof(projectTree));
+
+            return TryMove(project, projectTree, MoveAction.Down);
+        }
+
+        /// <summary>
+        /// Move the project tree down over one of its siblings.
+        /// </summary>
+        public static Task<bool> TryMoveDownAsync(ConfiguredProject configuredProject, IProjectTree projectTree)
+        {
+            Requires.NotNull(configuredProject, nameof(configuredProject));
+            Requires.NotNull(projectTree, nameof(projectTree));
+
+            return TryMoveAsync(configuredProject, projectTree, MoveAction.Down);
+        }
+
+        /// <summary>
+        /// Determines if we are moving up or down files or folders.
+        /// </summary>
+        private enum MoveAction { Up=0, Down=1 }
+
+        /// <summary>
+        /// Gets a read-only collection with the evaluated includes associated with a project tree.
+        /// </summary>
+        private static ReadOnlyCollection<string> GetEvaluatedIncludes(IProjectTree projectTree)
+        {
+            var treeQueue = new Queue<IProjectTree>();
+            var includes = new List<string>();
+
+            treeQueue.Enqueue(projectTree);
+
+            // The queue is how we process each project tree.
+            while (treeQueue.Count > 0)
+            {
+                var tree = treeQueue.Dequeue();
+
+                if (tree.IsFolder)
+                {
+                    foreach (var childTree in tree.Children)
+                    {
+                        treeQueue.Enqueue(childTree);
+                    }
+                }
+                else
+                {
+                    if (tree is IProjectItemTree2 tree2 && IsValidDisplayOrder(tree2.DisplayOrder))
+                    {
+                        includes.Add(tree2.Item.ItemName);
```

Comment:
This is unique enough?

Reply:
It's the evaluated include, which I assume so. Though, thinking about it, it might be possible there could be two, but it's unlikely.

Reply:
I need to do a little bit more work on this bit. Includes need to be in order.

Reply:
Ok, I have a sanity check for uniqueness, and the includes are now in order by their display order.